src/main/webapp/app/exercises/shared/result/result.service.ts

```diff
@@ -117,4 +123,39 @@ export class ResultService implements IResultService {
         }
         return participation;
     }
+
+    /**
+     * Fetches all results for an exercise and returns them
+     */
+    getResults(exercise: Exercise) {
+        return this.getResultsForExercise(exercise.id!, {
+            withSubmissions: exercise.type === ExerciseType.MODELING,
+        }).pipe(
+            tap((res: HttpResponse<Result[]>) => {
+                return res.body!.map((result) => {
+                    result.participation!.results = [result];
+                    (result.participation! as StudentParticipation).exercise = exercise;
+                    if (result.participation!.type === ParticipationType.PROGRAMMING) {
+                        addUserIndependentRepositoryUrl(result.participation!);
+                    }
+                    result.durationInMinutes = this.durationInMinutes(
+                        result.completionDate!,
+                        result.participation!.initializationDate ? result.participation!.initializationDate : exercise.releaseDate!,
+                    );
+                    // Nest submission into participation so that it is available for the result component
+                    if (result.participation && result.submission) {
+                        result.participation.submissions = [result.submission];
+                    }
+                    return result;
+                });
+            }),
+        );
+    }
+
+    /**
+     * Utility function
+     */
+    private durationInMinutes(completionDate: Moment, initializationDate: Moment) {
+        return this.momentDiff.transform(completionDate, initializationDate, 'minutes');
```

Comment:
You are already importing moment, is this pipe doing something that moment can't do?

Reply:
Just left it as it was before.. But refactoring it now using moment instead :+1: